src/core/qgsrelation.h

```diff
@@ -113,12 +113,13 @@ class CORE_EXPORT QgsRelation
     /**
      * Creates a relation from an XML structure. Used for reading .qgs projects.
      *
+     * \param project the project to get reference layers from
      * \param node The dom node containing the relation information
      * \param context to pass project translator
      *
      * \returns A relation
      */
-    static QgsRelation createFromXml( const QDomNode &node, QgsReadWriteContext &context );
+    static QgsRelation createFromXml( const QgsProject &project, const QDomNode &node, QgsReadWriteContext &context );
```

Comment:
This is an api break. Either create a new overload or add it as an optional parameter (which can be turned into a mandatory parameter in QGIS 4)

Reply:
Ok, I'm changing this to an optional pointer and fallback fetching project fron `QgsProject::instance()` when passed pointer is null.